libs/vkd3d/state.c

```diff
@@ -2332,6 +2334,12 @@ static uint32_t d3d12_graphics_pipeline_state_init_dynamic_state(const struct d3
         }
     }
 
+    /* We always need to enable fragment shading rate dynamic state when rasterizing.
+     * D3D12 has no information about this ahead of time for a pipeline
+     * unlike Vulkan */
+    if (d3d12_device_supports_variable_shading_rate_tier_1(state->device) && graphics->rt_count && graphics->sample_mask)
```

Comment:
Does D3D support imageless framebuffer, i.e. rendering without any framebuffer which relies on sideeffects?

Reply:
You can kinda do it with ForcedSampleCount for Target Independent Rasterization which we don't handle right now, but enabling VRS with that is undefined behaviour:

https://microsoft.github.io/DirectX-Specs/d3d/VariableRateShading.html
`TIR is not supported when subsampling is used.`

Reply:
Added a comment.